commons/src/main/java/io/aiven/kafka/tieredstorage/commons/storage/BytesRange.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.commons.storage;
+
+/**
+ * Byte range with from and to edges; where to cannot be less than from.
+ * inclusive/exclusive semantics are up to the consumer.
+ *
+ * @see FileFetcher#fetch(String, BytesRange)
+ * @see org.apache.kafka.server.log.remote.storage.RemoteStorageManager#fetchLogSegment(
```

Comment:
Do we need to have this reference here? In RSM interface there is no BytesRange so IMO it looks a little confusing. I would avoid having all this usage links altogether since the class + javadoc should be enough to understand how to use it IMO.